### PR TITLE
make package name red if firmware conflict

### PR DESCRIPTION
**Pull request by zachary7829**
makes the color of the package name red if an incompatible firmware is specified from the control file, makes browsing through packages on a repo more clear that it isn't supported.

### CODE REVIEW

**Comment by CFSerena** [comment]
Only firmware conflict? Is it possible to just checks for any firmware in general?

**Comment by devAgam** [comment]
> Only firmware conflicts? Is it possible to just checks for any conflicts in general?

I think that might be a little too much considering that most of the conflicts can be resolved by a simple uninstall or install of the dependency or conflict

File: Sileo/UI/PackageViewController/PackageViewController.swift [hunk 1]
```diff
@@ -691,6 +705,21 @@ class PackageViewController: SileoViewController, PackageQueueButtonDataProvider
...
+            if let rangeofendfirmware: Range<String.Index> = newconflicts.range(of: ")") {
+            var firmwareinconflict = String(newconflicts[(range2.lowerBound ..< rangeofendfirmware.lowerBound)])
+            firmwareinconflict = firmwareinconflict.replacingOccurrences(of: ">=",with: "")
+            let number = (firmwareinconflict as NSString).floatValue
+            return number
+            }
+        }
+    }
+        return 0.0
+    }
```
Comment (CFSerena) [inline Sileo/UI/PackageViewController/PackageViewController.swift:721]: Try this instead: 
```swift
func getFirmwareFromDepends(conflicts: String) -> Float {
    guard let range = conflicts.range(of: "firmware") else {
        return 0.0
    }
    let newConflicts = conflicts[(range.lowerBound ..< conflicts.endIndex)]
    
    guard let secondRange = newConflicts.range(of: ">="), let endRangeOfFirmware = newConflicts.range(of: ")") else {
        return 0.0
    }
    
    let firmwareInConflict = String(newConflicts[(secondRange.lowerBound ..< endRangeOfFirmware.lowerBound)])
        .replacingOccurrences(of: ">=", with: "")
    
    return Float(firmwareInConflict) ?? 0.0
}
```

Comment (zachary7829) [reply Sileo/UI/PackageViewController/PackageViewController.swift:783]: this doesn't work to my testing

File: Sileo/UI/PackageListViewController/PackageCell/PackageCollectionViewCell.swift [hunk 2]
```diff
@@ -471,17 +503,49 @@ extension PackageCollectionViewCell: SwipeCollectionViewCellDelegate {
...
+        guard let range = conflicts.range(of: "firmware") else {
+            return true
+        }
+        let newConflicts = conflicts[(range.lowerBound ..< conflicts.endIndex)]
+        
+        guard let secondRange = newConflicts.range(of: "("), let endRangeOfFirmware = newConflicts.range(of: ")") else {
+            return true
+        }
+        
+        if (String(newConflicts[(secondRange.lowerBound ..< endRangeOfFirmware.lowerBound)]).contains("=")) {
```
Comment (CFSerena) [inline Sileo/UI/PackageListViewController/PackageCell/PackageCollectionViewCell.swift:545]: Scrap this and do `return String(newConflicts[(secondRange.lowerBound ..< endRangeOfFirmware.lowerBound)]).contains("=")` instead

File: Sileo/UI/PackageViewController/PackageViewController.swift (hunk 1 above)
Comment (CFSerena) [reply Sileo/UI/PackageViewController/PackageViewController.swift:783]: loss

File: Sileo/UI/PackageListViewController/PackageCell/PackageCollectionViewCell.swift (hunk 2 above)
Comment (zachary7829) [reply Sileo/UI/PackageListViewController/PackageCell/PackageCollectionViewCell.swift:545]: just tested, and it works. any more requests or is rest good?

**Comment by CRKatri** [comment]
`deb`s use `>>` and `<<`, not `<` and `>`.

Also this does not correctly handle `Depends: firmware (>= 14.0) | libexpat1` correctly (from Procursus `git`).

File: Sileo/UI/PackageListViewController/PackageCell/PackageCollectionViewCell.swift (hunk 2 above)
Comment (CFSerena) [reply Sileo/UI/PackageListViewController/PackageCell/PackageCollectionViewCell.swift:545]: Tons more requests but I’m not home atm, for one, the code looks super performance intensive

**Comment by zachary7829** [comment]
> `deb`s use `>>` and `<<`, not `<` and `>`.
> 
> Also this does not correctly handle `Depends: firmware (>= 14.0) | libexpat1` correctly (from Procursus `git`).

should be fixed now

**Pull request closed without merging**